src/components/Checkbox/Checkbox.tsx

```diff
@@ -42,7 +42,11 @@ export interface CheckboxProps {
   /** Display an error message */
   error?: Error | boolean;
   /** Callback when checkbox is toggled */
-  onChange?(newChecked: boolean, id: string): void;
+  onChange?(
+    newChecked: boolean,
+    id: string,
+    event?: React.KeyboardEvent | React.MouseEvent<Element, MouseEvent>,
```

Comment:
Do we need to pass the event here? There have been ongoing discussions about this and we have always opted to not pass the event. cc @kyledurand

Reply:
@kyledurand @bpscott there was discussion in this repo https://github.com/orgs/Shopify/teams/polaris/discussions/ but since the team name changed I can't find it.

Reply:
Fore more context here is the reasons why I'm passing the event inside the `onChange` handler.

Inside the `onInteraction` method of the `RowContext` requires this event as arg.  It needs to ensure that we stop the propagation of the event further up in the DOM tree and avoids unwanted side-effects.  Although I'm not a fan of always passing event through javascript method args, I feel like in this case it's necessary to avoid the need of further refactoring in the IndexTable.  

Also the way the code was written before my proposal,  we had a `<div>` wrapper around the `Checkbox` component which would listen to the `onClick` and `onKeyUp` events that were bubbling up. With this in mind, it's safe to say that passing the event through the method args serves the same purpose.

Reply:
Thanks @BPScott for the context.

After reading and gathering context, I went back to the code and tried to see how I could avoid adding the event in our API while still properly supporting the event bubbling.

**The fix**
 - On the wrapping div inside the `IndexTable/Checkbox`, we only now only listen to the bubbling `onClick` event.
    https://github.com/Shopify/polaris-react/blob/dfb406c6e4f75b751cbd4b3909ab3471bf896658/src/components/IndexTable/components/Checkbox/Checkbox.tsx#L36-L39

 - Inside the `Polaris/Checkbox` component, we stop the propagation of `onClick` and `onKeyUp` events emerging from the `spin.backdrop` DOM element 
    https://github.com/Shopify/polaris-react/blob/dfb406c6e4f75b751cbd4b3909ab3471bf896658/src/components/Checkbox/Checkbox.tsx#L180-L184

**The explanation**
To be able to bind on the `div.onClick` on the wrapper we need to make sure that only the click events coming from the `<Input />` control are bubbled up. Also since the activation (space for a checkbox) of a control will trigger the `onClick` event we need to ignore the `onKeyUp` events bubbled up to the wrapping div.